sky/spot/recovery_strategy.py

```diff
@@ -327,9 +327,42 @@ def _launch(self,
                     return None
                 logger.info('Failed to launch the spot cluster with error: '
                             f'{common_utils.format_exception(e)})')
+            except exceptions.FileMountError as e:
+                # If the file mounts fails, the failure can be due to:
+                # 1. Errors before the job submission, e.g., invalid file mounts
+                #   -- user changed the bucket content on the cloud manually
+                #   during the launch/recovery. We should not retry the launch,
+                #   as the error is not recoverable.
+                # 2. Preemption during the file mounts. We should be able to
+                #   find the cluster not in UP status, and we can retry the
+                #   launch.
+                #
+                # Pull the actual cluster status from the cloud provider to
+                # determine whether the cluster is preempted.
+                time.sleep(5)
+                (cluster_status,
+                 _) = backend_utils.refresh_cluster_status_handle(
+                     self.cluster_name,
+                     force_refresh_statuses=set(status_lib.ClusterStatus))
+                if cluster_status == status_lib.ClusterStatus.UP:
+                    # Case 1: Errors before the job submission. We should fail
+                    # fast and not retry the launch.
+                    # Note: the underlying file mounts will retry for the
+                    # network glitches, so we do not need to retry here.
+                    logger.info('User file mounts failed. Not retrying.\n'
+                                '  Detailed exception: '
+                                f'{common_utils.format_exception(e)}')
+                    # We should raise the exception, no matter whether
+                    # raise_on_failure is True or False, because even though the
+                    # user file mounts suceeded in the previous launch, contents
+                    # in the bucket may have changed, and we should not retry
+                    # the launch, if the user file mounts failed.
+                    raise
```

Comment:
Update docstring for this function with new error types?

Reply:
Also, update docstring for `execution.py::launch`?

Reply:
Good catch! Fixed. Thanks!